rfc/0011-value-transformer/README.md

```diff
@@ -0,0 +1,191 @@
+  output hundreds oftype integer;
+
+  ones: number % 10;
+  tens: floor (number / 10) % 10;
+  hundreds: floor (number / 100);
+}
+```
+
+### Usage 1: TableInterpreter
+
+The `TableInterpreter` block uses a transformer from `text` to the desired value type with the `with` keyword. Parsing can still be used as currently (`as` keyword)
+
+Example:
+```
+block GasReserveTableInterpreter oftype TableInterpreter {
+  header: true;
+  columns: [ // only allow transformers with one input and one output or plain parsing here
+      "Datum" with DateTransformer,
+      "Kritisch" with PercentTransformer,
+      "Angespannt" as text,
+      "Stabil" with GermanToBooleanTransformer,
+      "Speicherstand IST" with PercentTransformer,
+      "gesetzliche Ziele" with PercentTransformer
+    ];
+}
+```
+
+This version only works with transformers that have exactly one input and exactly one output. We might need to come up with a different mechanism to allow other transformers (requires future RFCs).
+
+
+### Usage 2: TableTransformerBlock
+There is a newly introduced `TableTransformerBlock` block type that applies the mapper to a column in a `table`.
+
+Example:
+```
+block GermanToBooleanBlock oftype TableTransformerBlock {
+  inputColumns: ["Kritisch"];  // array to satisfy all inputs of transformer
+  outputColumns: ["IsCritical"]; // might also overwrite one of the input columns
+  use: GermanToBooleanTransformer;
+}
+```
+
+### Usage 3: SheetTransformerBlock
+There is a newly introduced `SheetTransformerBlock` block type that applies the mapper to a set of cells in a `sheet`. Please note that the value type is not stored in sheets and thus lost after computation. Outputs are therefore parsed to string automatically by the block.
+
+Example:
+```
+block GermanToBooleanBlock oftype SheetTransformerBlock {
+  inputCells: [range A3:A17]; // if multiple arguments: have to match in dimensions
+  outputCells: [range D3:D17]; // has to match dimension of inputs
+  use: GermanToBooleanTransformer;
+}
+```
+
+
+## Drawbacks
+- declaration of in and outputs might be much boilerplate.
+- TableInterpreter blocks only allow transformers with one input and one output.
+
+## Alternatives
+
+- use other keyword, e.g., `vtrans`, `mapper`, ...
+- shorten `input` and `output` keywords to `in` and `out`
+- Transformers cannot work on sheets
+- design nicer syntax for multiple inputs and outputs via names instead of relying on the index
+- refactor TableInterpreter completely to also allow transformers with multiple in and outputs
+
+## Possible Future Changes/Enhancements
+
+- We could introduce a pipe-like syntax that allows chaining transformers.
+
+### Variables in Transformers
+Storing interim results can improve coding experience.
+```
+transformer <name> {
+  // inputs and outputs
+  var <varBane> oftype <valuetype>: <expression producing valuetype>; // alternative: infer type automatically 
+  // use variable to compute outputs
+}
+```
+
+### Composite ValueTypes
```

Comment:
I think this overall makes sense, the suggestion kind of naturally extends to composite valuetypes.